moped-database/migrations/1667853991173_add_created_by_sql_view/up.sql

```diff
@@ -0,0 +1,188 @@
+DROP VIEW project_list_view;
+
+CREATE OR REPLACE VIEW public.project_list_view
+AS WITH project_person_list_lookup AS (
+    SELECT
+      mpp.project_id,
+      string_agg(DISTINCT concat(mu.first_name, ' ', mu.last_name, ':', mpr.project_role_name), ','::text) AS project_team_members
+    FROM moped_proj_personnel mpp
+      JOIN moped_users mu ON mpp.user_id = mu.user_id
+      JOIN moped_proj_personnel_roles mppr ON mpp.project_personnel_id = mppr.project_personnel_id
+      JOIN moped_project_roles mpr ON mppr.project_role_id = mpr.project_role_id
+    WHERE mpp.is_deleted = false
+      AND mppr.is_deleted = false
+    GROUP BY mpp.project_id
+  ), funding_sources_lookup AS (
+    SELECT 
+      mpf_1.project_id,
+      string_agg(mfs.funding_source_name, ', '::text) AS funding_source_name
+    FROM moped_proj_funding mpf_1
+      LEFT JOIN moped_fund_sources mfs ON mpf_1.funding_source_id = mfs.funding_source_id
+    WHERE mpf_1.is_deleted = false
+    GROUP BY mpf_1.project_id
+  ), project_type_lookup AS (
+    SELECT
+      mpt.project_id,
+      string_agg(mt.type_name, ', '::text) AS type_name
+      FROM moped_project_types mpt
+        LEFT JOIN moped_types mt ON mpt.project_type_id = mt.type_id AND mpt.is_deleted = false
+    GROUP BY mpt.project_id
+  ),
+   added_by_lookup AS (
+    SELECT
+      concat(users.first_name, ' ', users.last_name) AS added_by,
+      project_id
+      FROM moped_project
+        LEFT JOIN moped_users users ON users.user_id = moped_project.added_by
+    )
```

Comment:
i'm not sure if this matters—this query is way beyond my SQL comfort zone—but because we're joining the one (a user) in a many-to-one relationship, i don't think we need a subquery. you can concat directly against moped_users.

```sql
     SELECT ...
     CONCAT(added_by_user.first_name, ' ', added_by_user.last_name) AS added_by_user
     ...
     FROM LEFT JOIN moped_users added_by_user ON mp.added_by = added_by_user.user_id
```

i would only be concerned about this if there's a performance hit for using a subquery

Reply:
yup, i overengineered this. ill update it